ddtrace/tracer/remote_config.go

```diff
@@ -158,18 +158,32 @@ func (t *tracer) onRemoteConfigUpdate(u remoteconfig.ProductUpdate) map[string]s
 	return statuses
 }
 
-// startRemoteConfig starts the remote config client
-// and registers the APM_TRACING product and its callback.
+// startRemoteConfig starts the remote config client.
+// It registers the APM_TRACING product with a callback,
+// and the LIVE_DEBUGGING product without a callback.
 func (t *tracer) startRemoteConfig(rcConfig remoteconfig.ClientConfig) error {
 	err := remoteconfig.Start(rcConfig)
 	if err != nil {
 		return err
 	}
-	return remoteconfig.Subscribe(
+
+	if t.config.dynamicInstrumentationEnabled {
+		err = remoteconfig.RegisterProduct(state.ProductLiveDebugging)
+		if err != nil {
+			return err
+		}
```

Comment:
⚠️ I don't know when/why `RegisterProduct` might fail, but I suspect we might still want to try to register `ProductAPMTracing` below when that happens?

Reply:
Agree, this shouldn't affect anything else. 

If registering our product fails we should just log an error and let the program continue.

Reply:
That sounds right to me. What do you think of the change I made here? The error return from `startRemoteConfig` get's logged.

Reply:
I think it's fine now - we could ignore the DI error and just return the APM result to keep the previous behaviour exactly as it was but returning error when either fail is also fine; to get a DI error users would need to set the env var.